presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -48,6 +49,8 @@ public ZOrder(List<Integer> encodingBits)
 
         totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum();
         maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max().getAsInt();
+
+        curveSearcher = new ZOrderCurveSearcher(this);
```

Comment:
If we need `ZOrder` field access in `ZOrderCurveSearcher`, we should pass individual fields into `ZOrderCurveSearcher`